tasks/milmove.py

```diff
@@ -27,11 +35,306 @@ def on_start(self):
+        easier to see in the logs what is happening.
+        """
+        user = self.get_user_info()
+
+        if user is None:
+            return  # get_user_info will log the error
+
+        service_member_id = user["service_member"]["id"]
+
+        milmove_faker = MilMoveData()
+
+        # First let's create the profile.
+        payload = {
+            "id": service_member_id,
+            "affiliation": "ARMY",
+            "edipi": "1234567890",
+            "rank": "E_1",
+        }
+
+        sm_url, sm_request_kwargs = self.request_preparer.prep_internal_request(
+            endpoint=f"/service_members/{service_member_id}",
+            endpoint_name="/service_members/{serviceMemberId}",
+        )
+
+        with self.rest(method="PATCH", url=sm_url, data=json.dumps(payload), **sm_request_kwargs) as resp:
+            resp: RestResponseContextManager
+
+            task_name = "create_user_profile"
+
+            log_response_info(response=resp, task_name=task_name)
+
+            if resp.status_code != HTTPStatus.OK:
+                resp.failure("Unable to create customer profile.")
+
+                log_response_failure(response=resp, task_name=task_name)
+
+                return
+
+        # Now we can add name info. The url and request_kwargs will actually be the same, so we only
+        # need to prepare the payload.
+
+        payload = {
+            "id": service_member_id,
+            "first_name": milmove_faker.get_fake_data_for_type(data_type=DataType.FIRST_NAME),
+            "middle_name": milmove_faker.get_fake_data_for_type(data_type=DataType.FIRST_NAME),
+            "last_name": milmove_faker.get_fake_data_for_type(data_type=DataType.LAST_NAME),
+            "suffix": "",
+        }
+
+        with self.rest(method="PATCH", url=sm_url, data=json.dumps(payload), **sm_request_kwargs) as resp:
+            resp: RestResponseContextManager
+
+            task_name = "add_name_info"
+
+            log_response_info(response=resp, task_name=task_name)
+
+            if resp.status_code != HTTPStatus.OK:
+                resp.failure("Unable to add customer name.")
+
+                log_response_failure(response=resp, task_name=task_name)
+
+                return
+
+        # Now we can add contact info. Again, url and request_kwargs will be the same, so only need
+        # a payload.
+
+        payload = {
+            "id": service_member_id,
+            "telephone": milmove_faker.get_fake_data_for_type(data_type=DataType.PHONE),
+            "secondary_telephone": milmove_faker.get_fake_data_for_type(data_type=DataType.PHONE),
+            "personal_email": milmove_faker.get_fake_data_for_type(data_type=DataType.EMAIL),
+            "email_is_preferred": milmove_faker.get_fake_data_for_type(data_type=DataType.BOOLEAN),
+            "phone_is_preferred": milmove_faker.get_fake_data_for_type(data_type=DataType.BOOLEAN),
+        }
```

Comment:
I think it's fine to override and remove fields to simulate a partial patch but we're not getting the benefit of the generate_fake_request that sometimes sets optional fields and sometimes doesn't here and other places like middle_name and suffix.

Reply:
Ah good point, I had gone back and added some of the optional fields on a second pass through, but did forget to define a value for the suffiix. And in general, yeah, this is definitely a downside to using this data generator so directly.